Add tests

tacaswell opened the pull request.

tacaswell committed:
MNT : add api level for database objects

precursor work to possibly collapsing mds.database package to a
single module

Commit:
ENH: Added beamline_id to Event document template

Commit:
FIX: Fixed minor typo in example/collection.py

Commit:
ENH: Added search_events_broker() to commands for broker to be able to pull data given event search parameters

tacaswell committed:
PEP8 : whitespace

tacaswell committed:
TST : add test skeleton + test for save_beamline_config

tacaswell committed:
DBA : make descriptor_name not unique

tacaswell committed:
TST : add travis integration

ericdill commented on metadataStore/test/test_commands.py:
```diff
@@ -0,0 +1,58 @@
+
+def context_decorator(func):
+    def inner(*args, **kwargs):
+        with switch_db(BeamlineConfig, 'test_db'), \
```
use `()` to get rid of the `\`

tacaswell replied:
For reasons that I don't recall right now, you can't do that with context managers.  This use case is the only reason `\` line continuation has not been deprecated.

ericdill replied:
ah well fair enough

ericdill commented on .travis.yml:
```diff
@@ -0,0 +1,29 @@
+  - python run_tests.py
+
+after_success:
+  coveralls
```
hah, you already have the basics of coveralls ready for me

ericdill committed:
ENH: First attempt to add coveralls

tacaswell committed:
MNT/TST/API : updated event_descriptor

 - changed data_keys to a dict to match standard document
 - simplified '.' <-> '[dot]' code
 - added tests

ericdill committed:
ENH: Added badges

tacaswell committed:
DOC : formatted/corrected docs

tacaswell committed:
PEP8/DOC : white space fixes

tacaswell committed:
Merge remote-tracking branch 'ericdill/add_coverage' into add_tests

tacaswell committed:
DOC : add optional flags

tacaswell committed:
MNT : simplify create_time search handling logic

tacaswell committed:
TST : return the created documents

maybe useful later

Commit:
MNT: Add test subpackage to test ignoring

tacaswell committed:
Merge pull request #1 from ericdill/add_coverage

MNT: Add test subpackage to test ignoring
